packages/block-library/src/query-no-results/index.php

```diff
@@ -33,7 +47,13 @@ function render_block_core_query_no_results( $attributes, $content, $block ) {
 		$query = $wp_query;
 	} else {
 		$query_args = build_query_vars_from_query_block( $block, $page );
-		$query      = new WP_Query( $query_args );
+
+		// Add search parameter if instant search is enabled and search query exists
+		if ( $enhanced_pagination && $instant_search_enabled && ! empty( $search_query_direct ) ) {
+			$query_args['s'] = $search_query_direct;
+		}
+
+		$query = new WP_Query( $query_args );
```

Comment:
Why do we need to create a new Query (and do so multiple times) instead of updating the current one?

Reply:
I'm not sure I understand what you mean. 

We're not adding (another) new Query here. The code of the `Query No Results` block was already creating a `new WP_Query()` (line 36) - we're just passing the search parameter to it.

Reply:
You'll notice that in other blocks modified in this PR ([`post template`](https://github.com/WordPress/gutenberg/pull/63147/files/1b7b4bff8f1c528180652cf5d4eef8efd169fd4b#diff-4657bc4ddd88ab4cf8b48ad35b2fcf177edd40040c5976e2a3f9b89b52dd2bc7R91), [`query-pagination-next`](https://github.com/WordPress/gutenberg/pull/63147/files/1b7b4bff8f1c528180652cf5d4eef8efd169fd4b#diff-d58528784de1473c7d03da21f7c2cd742e8b36eefe09a252bb39d384ed892b39R75), or [`query-pagination-numbers`](https://github.com/WordPress/gutenberg/pull/63147/files/1b7b4bff8f1c528180652cf5d4eef8efd169fd4b#diff-14fbad12ea000a511336bf91b560cb395b654f881060942f2b4495f0bb16770bR66) we're also not creating any additional queries beyond what had existed before. In `query-pagination-next` and `query-pagination-numbers` at most one query will run - which one depends on whether Instant Search is enabled.

Reply:
Oh, [I see](https://github.com/WordPress/gutenberg/blob/b867a5fe7c1133e906682eff84b53b6621dea559/packages/block-library/src/query-pagination-next/index.php#L57C43-L57C76).